BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/signal/WindowFilter.java

```diff
@@ -8,28 +8,37 @@
 
 public abstract class WindowFilter implements RssiFilter {
 
-    protected static final long DURATION_DEFAULT = TimeUnit.SECONDS.toMillis(3);
+    public static long DEFAULT_DURATION = TimeUnit.SECONDS.toMillis(5);
 
-    protected long duration = DURATION_DEFAULT;
-    protected long maximumTimestamp = System.currentTimeMillis();
-    protected long minimumTimestamp = maximumTimestamp - DURATION_DEFAULT;
+    protected long duration = DEFAULT_DURATION;
+    protected long maximumTimestamp;
+    protected long minimumTimestamp;
+    protected TimeUnit timeUnit;
 
-    public WindowFilter(long duration, TimeUnit timeUnit){
+    public WindowFilter() {
+        this(DEFAULT_DURATION, TimeUnit.MILLISECONDS);
+    }
+
+    public WindowFilter(long duration, TimeUnit timeUnit) {
         this.minimumTimestamp = this.maximumTimestamp - timeUnit.toMillis(duration);
```

Comment:
No `maximumTimestamp` available at this moment. Therefore you will have `maximumTimestamp == 0` and `minimumTimestamp == - timeUnit.toMillis(duration)`. Also there is no need to access them with `this` if you don't have local variables with the same name.